packages/app/src/webapp/models/table.ts

```diff
@@ -103,16 +106,31 @@ export class Table {
   }
 }
 
-type MultiTable = Table[]
+export interface WatchableTable extends Table, IWatchable {}
+
+export function formatWatchableTable (model: Table | Table[], watch: IWatchable) {
+  if (isTable(model)) {
+    return Object.assign(model, watch)
+  } else if (isMultiTable(model)) {
+    model.forEach(table => Object.assign(table, watch))
+  } else {
+    // TODO: we might need to consider the variance of model, throw error for now
+    throw new Error('models other than table(s) are not supported in watch mode yet')
+  }
+}
 
 export function isTable (model: any): model is Table {
   return model !== undefined && (model instanceof Table || (model as Table).body !== undefined)
 }
 
-export function isMultiTable (model: any): model is MultiTable {
+export function isMultiTable (model: any): model is Table[] {
   return model !== undefined && Array.isArray(model) && model.length > 0 && model.filter(m => !isTable(m)).length === 0
 }
 
+export function isWatchable (model: WatchableTable): model is WatchableTable {
```

Comment:
this should accept Table|WatchableTable; this will allow you to remove all of the "as WatchableTable" typecasts in the table.ts code. please try to limit these kinds of typecasts, as they circumvent static type safety